### DISCUSSION TITLE
Custom handler to process images before display

### DISCUSSION THREAD

**Discussion by feklee** [filebrowser/filebrowser#3465] [Q&A]
An example of an image format that needs processing:

[_DSC4973.ARW.zip](https://github.com/user-attachments/files/16902510/_DSC4973.ARW.zip)

It’s a photo in raw format that I shot just this morning during a T8 signal in Hong Kong, empty streets. No web browser known to me can display `.ARW` files. However, it is trivial to extract a preview JPEG embedded in a `.ARW` file:

    exiftool -b -ThumbnailImage _DSC4973.ARW >preview.jpg

Or the thumbnail:

    exiftool -b -Thumbnailimage _DSC4973.ARW >thumb.jpg

*Is it possible to tell Filebrowser to do as above, then show the JPEG as a preview for the file?*

Some kind of custom processor for handling file formats not known to web browsers would come in handy with other formats as well. There are tons of custom image formats, especially when it comes to 2D vector formats or 3D formats.

For comparison: The Dropbox web based file browser does show a preview for raw image files. (but I stopped using Dropbox yesterday, swapped it for Syncthing).

**Comment by nobeltnium** [comment]
Having the same problem right now. I'm also using ARW format from Sony camera.
I have to cheat it abit. Using SMB sharing so that I could open the directory containing the pictures, then use xnviewmp to view them through that.

But having the ARW viewer implement along with mp4 thumbnail could be a life saver FR
